internal/engine/actions.go

```diff
@@ -93,18 +93,12 @@ func NewBuildCompleteAction(result store.BuildResultSet, err error) BuildComplet
 
 type InitAction struct {
```

Comment:
this pr makes me wonder if InitAction should be an empty struct. I feel like a lot of these fields are just arguments and flags that would be easier to handle with dependency injection